javascript/appointment/appointmentEvent.js

```diff
@@ -0,0 +1,174 @@
+// Send axios request when the page is loaded, and do initialization for the form
+document.addEventListener('DOMContentLoaded', function () {
+
+  // To check the user's role, then decide want to display or hide the charts
+  axios.get('http://localhost:6500/api/appointments/isAdmin')
+    .then((response) => {
+      const isAdmin = response.data;
+
+      appointmentChart_displayCharts(isAdmin);  // To show or hide the charts
+
+    }).catch((err) => {
+      console.log('Error when checking is admin: ', err);
+  });
+  
+  // To get the appointments, display them in charts and calendar
+  axios.get('http://localhost:6500/api/appointments')
+    .then((response) => {
+      const { username, isAdmin, appointments } = response.data;
+
+      // If the user is admin, then create the charts and calendar
+      if(isAdmin){
+        appointmentChart_createPieChart('appointment-admin-pieChart', appointments);                    // Create pie chart
+        appointmentChart_createAreaChart('appointment-admin-areaChart', appointments);                  // Create area chart
+        appointmentCalendar_createCalendar('appointment-admin-calendar', appointments, username, isAdmin);  // Create calendar
+      }
+      // Else, only create the calendar
+      else{
+        appointmentCalendar_createCalendar('appointment-admin-calendar', appointments, username, isAdmin);  // Create calendar
+      }
+
+      endLoader();
+
+    }).catch((err) => {
+      console.log('Error when first initializing data: ', err);
+  });
+
+
+  // To send get request to get the attendee options list from server except the user himself/herself
+  axios.get('http://localhost:6500/api/appointments/userlist')
```

Comment:
jieyii help to change this to baseUrl var so that easy to change backend url in future